eng/SourceBuild.props

```diff
@@ -0,0 +1,27 @@
+<Project>
+
+  <PropertyGroup>
+    <GitHubRepositoryName>diagnostics</GitHubRepositoryName>
+    <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(ArcadeBuildFromSource)' == 'true' ">
```

Comment:
Looking for feedback on where and how I chose to do this.  There is a conditional project to build already in the [build.props](https://github.com/dotnet/diagnostics/blob/master/eng/Build.props).  Since this is SB specific, I felt it was best placed here.

Reply:
Sometimes projects put this in their main build file (`Build.proj`/`Build.props`) guarded by `DotNetBuildFromSource`:

https://github.com/dotnet/runtime/blob/cdff28e8c48097721293fd8436f059cc2c3c1b39/Build.proj#L20-L21

Reply:
Per feedback from @dseefeld, I removed the condition because the props file itself is conditionally included.  

My reasoning for including it here was because it is SB specific.  I felt it was valuable to encapsulate the SB specific properties here in the SB.props file.